src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -16,24 +16,22 @@ namespace Microsoft.OData.Core
 #if ODATALIB_ASYNC
     using System.Threading.Tasks;
 #endif
+
     #endregion Namespaces
 
     /// <summary>
-    /// Class for reading OData batch messages; also verifies the proper sequence of read calls on the reader.
+    /// Abstract class for reading OData batch messages; also verifies the proper sequence of read calls on the reader.
     /// </summary>
-    public sealed class ODataBatchReader : IODataBatchOperationListener
+    public abstract class ODataBatchReader : IODataBatchOperationListener
     {
         /// <summary>The input context to read the content from.</summary>
-        private readonly ODataRawInputContext inputContext;
-
-        /// <summary>The batch stream used by the batch reader to devide a batch payload into parts.</summary>
-        private readonly ODataBatchReaderStream batchStream;
+        private ODataInputContext inputContext;
 
         /// <summary>True if the writer was created for synchronous operation; false for asynchronous.</summary>
         private readonly bool synchronous;
 
         /// <summary>The batch-specific URL resolver that stores the content IDs found in a changeset and supports resolving cross-referencing URLs.</summary>
-        private readonly ODataBatchUrlResolver urlResolver;
+        internal readonly ODataBatchUrlResolver urlResolver;

```

Comment:
Cannot be protected: class ODataBatchUrlResolver is internal, whose accessibility is more restricted than protected. Usually library should apply restrictive scope (such as "internal" here) as much as possible to prevent misuse from client, right?